src/WebJobs.Script.WebHost/Management/InstanceManager.cs

```diff
@@ -130,9 +160,20 @@ public async Task<string> ValidateContext(HostAssignmentContext assignmentContex
             HttpResponseMessage response = null;
             try
             {
+                _logger.LogInformation($"Will be using {assignmentContext.ZipUrlEnvVar} app setting as zip url");
+
                 var zipUrl = assignmentContext.ZipUrl;
                 if (!string.IsNullOrEmpty(zipUrl))
                 {
+                    // ScmRunFromPackage can be empty since it is a placeholder for the actual build zip
```

Comment:
Continuation of returning (string,string) - looks like ZipUrl can instead return HostAssignmentZipUrl object. Add a method IsValid() and return true for non scm urls and return actual value for scm url.